activerecord/lib/active_record/attribute_methods/serialization.rb

```diff
@@ -127,7 +127,7 @@ def serialize(attr_name, class_name_or_coder = Object, yaml: {}, **options)
             end
 
             cast_type = cast_type.subtype if Type::Serialized === cast_type
-            Type::Serialized.new(cast_type, coder)
+            Type::Serialized.new(cast_type, coder, default: columns_hash[attr_name.to_s]&.default)
```

Comment:
I missed something here. We can't access `columns_hash` from there because the app is still booting.

I'll try to find a solution, but if I can't I'll have to revert this. Sorry @serg-kovalev

Reply:
oh, good point, that's a surprise it's not covered by tests. I will try to reproduce in test and think about a fix as well.
Please keep me posted if you find a solution